src/solvers/refinement/string_refinement.cpp

```diff
@@ -1753,18 +2081,22 @@ exprt string_refinementt::get(const exprt &expr) const
 /// \param [out] witness: the witness of the satisfying assignment if one
 ///   exists. If UNSAT, then behaviour is undefined.
 /// \return: true if axiom is SAT, false if UNSAT
-bool string_refinementt::is_axiom_sat(
-  const exprt &axiom, const symbol_exprt& var, exprt &witness)
+static bool is_axiom_sat(
+  const namespacet& ns,
+  ui_message_handlert::uit ui,
+  const exprt &axiom,
+  const symbol_exprt& var,
+  exprt &witness)
```

Comment:
Could be made to return an optional `witness` instead of a bool and affecting the reference. But then for clarity it should be renamed something like `find_counter_example`